modules/navier_stokes/test/tests/finite_element/ins/energy-conservation/q2q1.i

```diff
@@ -0,0 +1,179 @@
+[Mesh]
+  [gen]
+    type = GeneratedMeshGenerator
+    nx = 10
+    ny = 10
+    dim = 2
+  []
+  [subdomain]
+    type = SubdomainBoundingBoxGenerator
+    bottom_left = '0.5 0 0'
+    top_right = '1 1 0'
+    block_id = 1
+    input = gen
+  []
+  [break_boundary]
+    input = subdomain
+    type = BreakBoundaryOnSubdomainGenerator
+    boundaries = 'bottom top'
+  []
+  [sideset]
+    type = SideSetsBetweenSubdomainsGenerator
+    input = break_boundary
+    primary_block = '1'
+    paired_block = '0'
+    new_boundary = 'fluid_left'
+  []
+  coord_type = RZ
+  second_order = true
+[]
+
+[Variables]
+  [T]
+    order = SECOND
+  []
+  [velocity]
+    family = LAGRANGE_VEC
+    order = SECOND
+    block = 1
+  []
+  [pressure]
+    block = 1
+  []
+[]
+
+[ICs]
+  [vel]
+    type = VectorConstantIC
+    variable = velocity
+    x_value = 1e-15
+    y_value = 1e-15
+    block = 1
+  []
+[]
+
+[Kernels]
+  [mass]
+    type = INSADMass
+    variable = pressure
+    block = 1
+  []
+
+  [momentum_convection]
+    type = INSADMomentumAdvection
+    variable = velocity
+    block = 1
+  []
+  [momentum_viscous]
+    type = INSADMomentumViscous
+    variable = velocity
+    block = 1
+  []
+  [momentum_pressure]
+    type = INSADMomentumPressure
+    variable = velocity
+    pressure = pressure
+    integrate_p_by_parts = true
+    block = 1
+  []
+  [momentum_supg]
+    type = INSADMomentumSUPG
+    variable = velocity
+    velocity = velocity
+    block = 1
+  []
+
+  [temperature_advection]
+    type = INSADEnergyAdvection
+    variable = T
+     block = 1
+  []
+  [temperature_supg]
+    type = INSADEnergySUPG
+    variable = T
+    velocity = velocity
+    block = 1
+  []
+  [temperature_conduction]
+    type = ADHeatConduction
+    variable = T
+    thermal_conductivity = 'k'
+  []
+  [heat_source]
+    type = BodyForce
+    variable = T
+    block = 0
+    function = 'x + y'
+  []
+  [remove_null_space]
+    variable = T
+    block = 0
+    type = Reaction
+  []
```

Comment:
you prefer this to setting a flux BC on temperature? this option would be more physical

Reply:
A flux BC on temperature does not remove the null space as the boundary flux terms only involve gradients of temperature (e.g. conduction). The advection term directly applies the incompressibility constraint and consequently does not perform integration by parts. This could be something we change in the future though

Reply:
with an initial condition the evolution of temperature should be well defined.

Reply:
like physically. numerically I can see your argument that there is nothing "holding" temperature only its gradient, that there is a null space

Reply:
I did end up going with an integrated BC, a convective heat flux BC on the right boundary. Nice suggestion for physicality!

Reply:
oh my bad. This is a steady problem so the initial condition isnt necessarily relevant